core/src/main/java/io/ultrabrew/metrics/data/BasicHistogramAggregator.java

```diff
@@ -0,0 +1,214 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the Apache License 2.0 license. See LICENSE file in Ultrabrew Metrics
+// for terms.
+
+package io.ultrabrew.metrics.data;
+
+import static io.ultrabrew.metrics.Metric.DEFAULT_CARDINALITY;
+import static io.ultrabrew.metrics.Metric.DEFAULT_MAX_CARDINALITY;
+
+import io.ultrabrew.metrics.Histogram;
+import io.ultrabrew.metrics.util.DistributionBucket;
+import java.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sun.misc.Unsafe;
+
+public class BasicHistogramAggregator extends ConcurrentIntTable implements Aggregator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(BasicHistogramAggregator.class);
+  private static final String[] AGG_FIELDS = {"count", "sum", "min", "max"};
+  private static final long[] IDENTITY = {0L, 0L, Long.MAX_VALUE, Long.MIN_VALUE};
+
+  private final String metricId;
+  private final DistributionBucket buckets;
+  private final int bucketCount;
+
+  public BasicHistogramAggregator(Histogram histogram) {
```

Comment:
javadocs everywhere

Reply:
This is a draft PR. Javadocs will follow. @smrutilal2 was trying to get feedback on whatever he had done so far while he's working on remaining parts.